quaddtype/numpy_quaddtype/_quaddtype_main.pyi

```diff
@@ -0,0 +1,134 @@
+    @property
+    @override
+    def kind(self) -> Literal["\x00"]: ...  # type: ignore[override]  # pyright: ignore[reportIncompatibleMethodOverride]
+    @property
+    @override
+    def num(self) -> Literal[-1]: ...  # type: ignore[override]  # pyright: ignore[reportIncompatibleMethodOverride]
+    @property
+    @override
+    def shape(self) -> tuple[()]: ...
+    @property
+    @override
+    def ndim(self) -> Literal[0]: ...
+    @property
+    @override
+    def fields(self) -> None: ...
+    @property
+    @override
+    def base(self) -> Self: ...
+    @property
+    @override
+    def subdtype(self) -> None: ...
+    @property
+    @override
+    def hasobject(self) -> Literal[False]: ...
+    @property
+    @override
+    def isbuiltin(self) -> Literal[0]: ...
+    @property
+    @override
+    def isnative(self) -> Literal[True]: ...
+    @property
+    @override
+    def isalignedstruct(self) -> Literal[False]: ...
+    @override
+    def __getitem__(self, key: Never, /) -> Self: ...  # type: ignore[override]
+
+@final
+class QuadPrecision:  # NOTE: It doesn't inherit from `np.generic` which is type-unsafe
+    def __new__(cls, /, value: _IntoQuad, backend: _Backend = "sleef") -> Self: ...
+
+    # Rich comparison operators
+    # NOTE: Unlike other numpy scalars, these return `builtins.bool`, not `np.bool`.
+    @override
+    def __eq__(self, other: object, /) -> bool: ...
+    @override
+    def __ne__(self, other: object, /) -> bool: ...
+    def __lt__(self, other: _IntoQuad, /) -> bool: ...
+    def __le__(self, other: _IntoQuad, /) -> bool: ...
+    def __gt__(self, other: _IntoQuad, /) -> bool: ...
+    def __ge__(self, other: _IntoQuad, /) -> bool: ...
+
+    # Binary operators
+    def __add__(self, other: _CastsQuad, /) -> Self: ...
+    def __radd__(self, other: _CastsQuad, /) -> Self: ...  # type: ignore[misc]
+    def __sub__(self, other: _CastsQuad, /) -> Self: ...
+    def __rsub__(self, other: _CastsQuad, /) -> Self: ...
+    def __mul__(self, other: _CastsQuad, /) -> Self: ...
+    def __rmul__(self, other: _CastsQuad, /) -> Self: ...  # type: ignore[misc]
+    def __pow__(self, other: _CastsQuad, mod: None = None, /) -> Self: ...
+    def __rpow__(self, other: _CastsQuad, mod: None = None, /) -> Self: ...
+    def __truediv__(self, other: _CastsQuad, /) -> Self: ...
+    def __rtruediv__(self, other: _CastsQuad, /) -> Self: ...
+
+    # Unary operators
+    def __neg__(self, /) -> Self: ...
+    def __pos__(self, /) -> Self: ...
+    def __abs__(self, /) -> Self: ...
+
+    # Conversion methods
+    def __bool__(self, /) -> bool: ...
+    def __int__(self, /) -> int: ...
+    def __float__(self, /) -> float: ...
+
+    # String representation
+    @override
+    def __repr__(self, /) -> str: ...
+    @override
+    def __str__(self, /) -> str: ...
+
+#
+def is_longdouble_128() -> bool: ...
+def get_sleef_constant(
```

Comment:
This is wrong - we need an override for the function since it returns normal integers for the integer constants bits and precision

Reply:
Ah yea of course; `"bits"` and `"precision"` return an `int`. I'll fix this in a follow-up.

And out of curiosity, why is this a function instead of "bare" constants?

Reply:
Constants can be re-exported 🤷🏻